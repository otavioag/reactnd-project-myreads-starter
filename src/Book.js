import React from 'react';
import BookMenu from './BookMenu';

const Book = (props) => (
  <li>
    <div className='book'>
      <div className='book-top'>
        <div
          className='book-cover'
          style={{
            width: 128,
            height: 193,
            backgroundImage: 'url('+(props.book.imageLinks ? props.book.imageLinks.thumbnail : '')+')'
          }}
        />
        <BookMenu book={props.book} shelf={props.book.shelf ? props.book.shelf : 'none'} handleShelfChange={props.handleShelfChange} />
      </div>
      <div className='book-title'>{props.book.title}</div>
      {props.book.authors ? props.book.authors.map(author => (
        <div key={author} className='book-authors'>{author}</div>
      )) : ''}
    </div>
  </li>
);

export default Book;