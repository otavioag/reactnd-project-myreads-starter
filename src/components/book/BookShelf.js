import React from 'react';
import PropTypes from 'prop-types';

const BookShelf = (props) => (
  <div className='bookshelf'>
    <h2 className='bookshelf-title'>{props.title}</h2>
    <div className='bookshelf-books'>
      <ol className='books-grid'>
        {props.books}
      </ol>
    </div>
  </div>
);

BookShelf.propTypes = {
  title: PropTypes.string.isRequired,
  books: PropTypes.array.isRequired
};

export default BookShelf;