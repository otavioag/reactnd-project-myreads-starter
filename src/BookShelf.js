import React from 'react';
import Book from './Book';

const BookShelf = (props) => (
  <div className='bookshelf'>
    <h2 className='bookshelf-title'>{props.title}</h2>
    <div className='bookshelf-books'>
      <ol className='books-grid'>
        {props.books.filter(book => book.shelf === props.shelf).map(book => (
          <Book
            key={book.id}
            book={book}
            handleShelfChange={props.handleShelfChange}
          />
        ))}
      </ol>
    </div>
  </div>
);

export default BookShelf;
