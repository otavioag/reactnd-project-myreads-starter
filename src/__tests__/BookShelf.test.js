import React from 'react';
import { shallow } from 'enzyme';
import BookShelf from '../components/book/BookShelf';
import Book from '../components/book/Book';

describe('BookShelf', () => {
  it('renders without crashing - with book data', () => {
    const wrapper = shallow(<BookShelf
      title=''
      books={[<Book
        key='key'
        book={{title: "React", authors: ["Nils Hartmann", "Oliver Zeigermann"], imageLinks: {thumbnail: "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"}, shelf: 'read'}}
        handleShelfChange={jest.fn()} />]}/>
    );
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});