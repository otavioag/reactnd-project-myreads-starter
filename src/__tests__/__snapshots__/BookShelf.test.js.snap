// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf renders without crashing - with book data 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  />
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <Book
        book={
          Object {
            "authors": Array [
              "Nils Hartmann",
              "Oliver Zeigermann",
            ],
            "imageLinks": Object {
              "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
            },
            "shelf": "read",
            "title": "React",
          }
        }
        handleShelfChange={[MockFunction]}
        key="key"
      />
    </ol>
  </div>
</div>
`;
